pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,172 @@
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return handleConfigMap(e.MetaNew)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+	}
+
+	// Watch for changes to the additional trust bundle configmap.
+	if err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, pred); err != nil {
+		return err
+	}
+
+	// Watch for changes to the proxy resource.
+	if err = c.Watch(&source.Kind{Type: &configv1.Proxy{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// handleConfigMap returns true if meta namespace is "openshift-config".
+// "openshift-config" is the namespace for one or more
+// ConfigMaps that contain user provided trusted CA bundles.
+func handleConfigMap(meta metav1.Object) bool {
+	return meta.GetNamespace() == "openshift-config"
+}
+
+var _ reconcile.Reconciler = &ReconcileProxyConfig{}
+
+// ReconcileProxyConfig reconciles a ClusterLogging object
+type ReconcileProxyConfig struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a cluslter-scoped
+// named "cluster" or a configmap object in namespace "openshift-config".
+func (r *ReconcileProxyConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	loggingNamespacedName := types.NamespacedName{Name: singletonName, Namespace: openshiftNS}
+	proxyNamespacedName := types.NamespacedName{Name: proxyName}
+	if request.NamespacedName == proxyNamespacedName {
+		proxyConfig := &configv1.Proxy{}
+		logrus.Debugf("Proxyconfig reconcile request.Name: '%s'", request.Name)
+		if err := r.client.Get(context.TODO(), request.NamespacedName, proxyConfig); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				logrus.Errorf("proxy not found; reconciliation will be skipped - request %v", request)
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, err
+		}
+	} else if request.Name == singletonName {
+		trustBundle := &corev1.ConfigMap{}
+		logrus.Debugf("Trust bundle configmap reconcile request.Namespace/request.Name: '%s/%s'", request.Namespace, request.Name)
+		if err := r.client.Get(context.TODO(), loggingNamespacedName, trustBundle); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				logrus.Errorf("configmap not found; reconciliation will be skipped - request %v", request)
```

Comment:
if we don't have a trust bundle anymore we should likely remove it from our components as well... (likely not an error message)

Reply:
> if we don't have a trust bundle anymore we should likely remove it from our components as well... (likely not an error message)

Currently, it is not removed when we see the trusted CA bundle is empty.  We calculate the hash and set it to the daemonset/deployment annotation.  When it's empty we keep the configmap with empty trusted CA bundle and set "0" to each annotation.
(please also note that it contains all the system trusted CA, so I'm wonder if there's the chance...)

Apart from it, I agree with the IsNotFound is not an error.  I'm removing the clause.